poly/src/evaluations/multivariate/multilinear/dense.rs

```diff
@@ -82,6 +99,64 @@ impl<F: Field> DenseMultilinearExtension<F> {
     pub fn iter_mut(&mut self) -> IterMut<'_, F> {
         self.evaluations.iter_mut()
     }
+
+    /// Concatenate the evaluation tables of multiple polynomials.
+    /// If the combined table size is not a power of two, pad the table with zeros.
+    ///
+    /// # Example
+    /// ```
+    /// use ark_test_curves::bls12_381::Fr;
+    /// use ark_poly::{MultilinearExtension, Polynomial, DenseMultilinearExtension};
+    /// use ark_ff::One;
+    ///
+    /// // Construct a 2-variate multilinear polynomial f1
+    /// // f1(x_0, x_1) = 2*(1-x_1)*(1-x_0) + 3*(1-x_1)*x_0 + 2*x_1*(1-x_0) + 6*x_1*x_0
+    /// let mle_1 = DenseMultilinearExtension::from_evaluations_vec(
+    ///     2, vec![2, 3, 2, 6].iter().map(|x| Fr::from(*x as u64)).collect()
+    /// );
+    /// // Construct another 2-variate MLE f2
+    /// // f2(x_0, x_1) = 1*x_1*x_0
+    /// let mle_2 = DenseMultilinearExtension::from_evaluations_vec(
+    ///   2, vec![0, 0, 0, 1].iter().map(|x| Fr::from(*x as u64)).collect()
+    /// );
+    /// let mle = DenseMultilinearExtension::merge(&[&mle_1, &mle_2]);
+    /// // The resulting polynomial is 3-variate:
+    /// // f3(x_0, x_1, x_2) = (1 - x_2)*f1(x_0, x_1) + x_2*f2(x_0, x_1)
+    /// // Evaluate it at a random point (1, 17, 3)
+    /// let point = vec![Fr::one(), Fr::from(17), Fr::from(3)];
+    /// let eval_1 = mle_1.evaluate(&point[..2].to_vec());
+    /// let eval_2 = mle_2.evaluate(&point[..2].to_vec());
+    /// let eval_combined = mle.evaluate(&point);
+    ///
+    /// assert_eq!(eval_combined, (Fr::one() - point[2]) * eval_1 + point[2] * eval_2);
+    pub fn merge(polys: impl IntoIterator<Item = impl AsRef<Self>> + Clone) -> Self {
```

Comment:
What is the use case for this function? When would I want to concatenate two polynomials?

Reply:
For example in the case of committing to polynomials, instead of committing to 2^r different x-variate polynomials, one can commit to their merged polynomial which is (r+x)-variate polynomial (which can be faster).

Reply:
This is something we could also use in our ark-spartan repo: https://github.com/arkworks-rs/spartan/blob/762fc57774fe70386532199911dd0955d58109e2/src/sparse_mlpoly.rs#L58

(will ofc require a larger refactor to switch from a custom `DensePoly` to one from `ark-poly`)

Reply:
Is there a nicer name for this? E.g. `concat`?

Reply:
Yes I agree `concat` would work better